src/p2p/net_node.inl

```diff
@@ -442,6 +446,46 @@ namespace nodetool
     return full_addrs;
   }
 
+  inline void assign_network_id(const boost::program_options::variables_map& vm, const bool testnet, Uuid& net_id)
+  {
+    Uuid id = ::config::NETWORK_ID;
+    std::string hint = "main-net's";
+
+    if(testnet)
+    {
+      if(command_line::has_arg(vm, arg_p2p_net_id))
+      {
+        const Uuid ext_id = command_line::get_arg(vm, arg_p2p_net_id);
+        if(ext_id != Uuid())
+        {
+          if(ext_id == ::config::NETWORK_ID)
+          {
+            std::ostringstream m;
+            m << "Fail on assigning network-id: an attempt to use main-net-network-id for test-net mode";
+            throw std::runtime_error(m.str());
+          }
+
+          if(ext_id == ::config::testnet::NETWORK_ID)
```

Comment:
do we really should rule this out?

Reply:
For sure it's not mandatory check. But if we have strange case I'd prefer to tell about it explicitly rather than hide it.